process-migration-service/src/main/java/org/kie/processmigration/service/impl/KieServiceImpl.java

```diff
@@ -139,17 +143,29 @@ private ProcessServicesClient getProcessServicesClient(String kieServerId) throw
         return kieServices.get(kieServerId).getServicesClient(ProcessServicesClient.class);
     }
 
-    private ProcessInfo getProcessInfo(String kieServerId, String containerId, String processId) throws InvalidKieServerException {
+    private ProcessInfo getProcessInfo(String kieServerId, String containerId, String processId) throws InvalidKieServerException, ProcessDefinitionNotFoundException {
         ProcessInfo processInfo = new ProcessInfo();
 
         //get SVG file
-        String svgFile = getUIServicesClient(kieServerId).getProcessImage(containerId, processId);
+        String svgFile;
+        try {
+            svgFile = getUIServicesClient(kieServerId).getProcessImage(containerId, processId);
+        } catch (KieServicesHttpException e) {
+            if(Response.Status.NOT_FOUND.getStatusCode() == e.getHttpCode()) {
+                throw new ProcessDefinitionNotFoundException(kieServerId, containerId, processId);
+            } else {
+                throw e;
+            }
+        }
 
         //Add this replacement here because in react-svgmt, ? and = are not allowed.
         svgFile = svgFile.replaceAll("\\?shapeType=BACKGROUND", "_shapeType_BACKGROUND");
         processInfo.setSvgFile(svgFile);
 
         ProcessDefinition pd = getProcessServicesClient(kieServerId).getProcessDefinition(containerId, processId);
+        if(!pd.getContainerId().equals(containerId)) {
+            throw new ProcessDefinitionNotFoundException(kieServerId, containerId, processId);
+        }
```

Comment:
Can't we just list containers available in KIE server and check if the container id from the input is among them?

Reply:
This is already reported as an enhancement https://issues.jboss.org/browse/JBPM-8473

Reply:
ok, thanks